### PR TITLE
the KCL can be configured with a DynamoDB client

### PR DESCRIPTION
**Pull request by danhodge**
Provides a mechanism that gives clients the ability to setup the KCL worker to talk to local DynamoDB instead of AWS, for example.

### CODE REVIEW

File: ext/src/main/java/com/kickstarter/jruby/Telekinesis.java [hunk 1]
```diff
@@ -45,11 +48,16 @@ public static Worker newWorker(final KinesisClientLibConfiguration config,
...
+        Worker.Builder builder = new Worker.Builder()
                 .recordProcessorFactory(v2Factory)
                 .config(config)
-                .execService(executor) // NOTE: .execService(null) is a no-op
-                .build();
+                .execService(executor); // NOTE: .execService(null) is a no-op
+
+        if (dynamoClient != null) {
+            builder.dynamoDBClient(dynamoClient);
+        }
```
Comment (blinsay) [inline ext/src/main/java/com/kickstarter/jruby/Telekinesis.java]: `.dynamoDBClient(null)` is the default value, so it's safe to not do this `if` check.

File: ext/src/main/java/com/kickstarter/jruby/Telekinesis.java [hunk 2]
```diff
@@ -33,10 +34,12 @@
 public class Telekinesis {
     /**
      * Create a new KCL {@link Worker} that processes records using the given
-     * {@link ExecutorService} and {@link IRecordProcessorFactory}.
+     * {@link ExecutorService} and {@link IRecordProcessorFactory}. Uses the
+     * given {@link AmazonDynamoDBClient} if non-null.
      */
     public static Worker newWorker(final KinesisClientLibConfiguration config,
                                    final ExecutorService executor,
+                                   final AmazonDynamoDBClient dynamoClient,
```
Comment (blinsay) [inline ext/src/main/java/com/kickstarter/jruby/Telekinesis.java]: I generally like to stick to the interface type where possible. can you make this an `AmazonDynamoDB` instead?

File: lib/telekinesis/consumer/kcl.rb [hunk 3]
```diff
@@ -23,6 +23,12 @@ class KCL
       # client in the same `:app` on the same host, make sure you set this to
       # something unique!).
       #
+      # Note: to configure the KCL clients to talk to a local DynamoDB server,
+      # pass in an instance of `AmazonDynamoDBClient` configured to talk to a
+      # local server as the second (optional) parameter to the KCL initialize
+      # method. If not provided, the KCL client will talk to AWS DynamoDB
+      # directly.
+      #
```
Comment (blinsay) [inline lib/telekinesis/consumer/kcl.rb]: mind rewording this so it's a little less specific? there are quite a few reasons I can see for passing a DynamoDB client and running a local dynamo doesn't seem like it'll come up that often.

i was thinking something along the lines of 

> Clients interested in configuring their own `AmazonDynamoDB` client may pass an instance as the second argument. If not configured, the client will use a default AWS configuration.

File: ext/src/main/java/com/kickstarter/jruby/Telekinesis.java (hunk 2 above)
Comment (danhodge) [inline ext/src/main/java/com/kickstarter/jruby/Telekinesis.java]: Good call, fixed: https://github.com/kickstarter/telekinesis/pull/12/commits/875bfc391780166e2ee87dd4f482e9b773ec2336

File: ext/src/main/java/com/kickstarter/jruby/Telekinesis.java (hunk 1 above)
Comment (danhodge) [inline ext/src/main/java/com/kickstarter/jruby/Telekinesis.java]: Nice catch: 8c6ffdb806fd9250438557e20d3168b0f82756e7

File: lib/telekinesis/consumer/kcl.rb [hunk 4]
```diff
@@ -64,10 +68,10 @@ class KCL
       #
       #     kcl_worker.run
       #
-      def initialize(config, &block)
+      def initialize(config, dynamo_client = null, &block)
```
Comment (blinsay) [inline lib/telekinesis/consumer/kcl.rb]: should this be `nil` and not `null`?

**Pull request merged**